yaksh/code_evaluator.py

```diff
@@ -61,9 +74,10 @@ def __init__(self, in_dir=None):
         msg = 'Code took more than %s seconds to run. You probably '\
               'have an infinite loop in your code.' % SERVER_TIMEOUT
         self.timeout_msg = msg
-        self.in_dir = in_dir
+        self.in_dir = in_dir if in_dir else MY_DIR
```

Comment:
Maybe the main `CodeEvaluator` should renamed to `Grader` or something like that.  Otherwise there are too many "Evaluators", there is `CodeEvaluator` and `BaseEvaluator` which are not related via inheritance which is confusing.